pkg/kcp/provider/gcp/iprange/syncAddress.go

```diff
@@ -28,20 +29,40 @@ func syncAddress(ctx context.Context, st composed.State) (error, context.Context
 	case client.ADD:
 		operation, err = state.computeClient.CreatePscIpRange(ctx, project, vpc, name, name, state.ipAddress, int64(state.prefix))
 	case client.MODIFY:
-		err = errors.New("IpRange update not supported.")
-		state.AddErrorCondition(ctx, v1beta1.ReasonNotSupported, err)
-		return composed.LogErrorAndReturn(err, "IpRange update not supported.", composed.StopAndForget, nil)
+		return composed.UpdateStatus(ipRange).
+			SetCondition(metav1.Condition{
+				Type:    v1beta1.ConditionTypeError,
+				Status:  metav1.ConditionTrue,
+				Reason:  v1beta1.ReasonNotSupported,
+				Message: "IpRange update not supported.",
+			}).
+			SuccessError(composed.StopAndForget).
```

Comment:
The removal of Ready condition is missing. 
I was thinking maybe we can have another UpdateStatus that by default knows we have two types and does the removals automatically i.e. remove the success when error is set and remove the error when ready is set. This will reduce the overhead and also prevents issues if we forget to do the removal correctly.
We can call it UpdateBinaryStatus or something like that. What do you think?

Reply:
Added `SetExclusiveCondition` method as discussed, and it will overwrite if there are any conditions before.